src/main/scala/wdltool/graph/WomGraphPrint.scala

```diff
@@ -0,0 +1,130 @@
+
+import scala.collection.JavaConverters._
+
+object WomGraphPrint {
+
+  final case class WorkflowDigraph(workflowName: String, digraph: NodesAndLinks)
+  final case class NodesAndLinks(nodes: Set[String], links: Set[String])
+
+  def readFile(filePath: String): String = Files.readAllLines(Paths.get(filePath)).asScala.mkString(System.lineSeparator())
+
+  def womExecutableFromWdl(filePath: String): Executable = {
+    val namespace = WdlNamespaceWithWorkflow.load(readFile(filePath), Seq(WdlNamespace.fileResolver _)).get
+    namespace.womExecutable
+  }
+
+  def womExecutableFromCwl(filePath: String): Executable = {
+    val yaml: Yaml = readFile(filePath)
+    CwlCodecs.decodeCwl(yaml) match {
+      case Right((clt: CommandLineTool, _)) =>
+        clt.womExecutable match {
+          case Valid(wom) => wom
+        }
+      case Right((wf: Workflow, m)) if m.isEmpty =>
+        wf.womExecutable(Map.empty) match {
+          case Valid(wom) => wom
+        }
+        // TODO: Don't throw away the error information! Might as well return it:
+      case other => throw new Exception(s"Can't read $filePath")
+    }
+  }
+
+  def generateWorkflowDigraph(file: String, auxFiles: Seq[String]): WorkflowDigraph = {
+
+    val womExecutable = if (file.toLowerCase().endsWith("wdl")) womExecutableFromWdl(file) else womExecutableFromCwl(file)
+    womExecutable.graph match {
+      case Valid(graph) =>
+        val digraph = listAllGraphNodes (graph)
+        WorkflowDigraph (dotSafe(womExecutable.entryPoint.name), digraph)
+      case Invalid(errors) =>
+        throw new IllegalArgumentException(errors.toList.mkString("\n", "\n", "\n"))
+    }
+  }
+
+  private val clusterCount: AtomicInteger = new AtomicInteger(0)
+
+  private def listAllGraphNodes(graph: Graph): NodesAndLinks = {
+
+    graph.nodes foldMap nodesAndLinks
+  }
+
+  private def nodesAndLinks(graphNode: GraphNode): NodesAndLinks = graphNode match {
+    case other => defaultNodesAndLinks(graphNode)
+  }
+
+  /**
+    * For most graph nodes, we draw a single box for the node, containing the name and input and output ports.
+    * For each input/output port, we include the links to upstream ports.
+    */
+  private def defaultNodesAndLinks(graphNode: GraphNode): NodesAndLinks = {
+    val clusterName = "cluster_" + clusterCount.getAndIncrement()
+    val portNodes = (graphNode.outputPorts ++ graphNode.inputPorts) map { p =>
+      s"${graphId(p)} [shape=${graphShape(p)} label=${graphName(p)}];"
+    }
+    val links = for {
+      inputPort <- graphNode.inputPorts
+      upstreamPort = inputPort.upstream
+    } yield graphId(upstreamPort)+" -> "+graphId(inputPort)
+
+    val singleNode =
+      s"""
+         |subgraph $clusterName {
+         |  style=filled;
+         |  fillcolor=${graphFillColor(graphNode)};
+         |  ${portNodes.mkString(sep="\n  ")}
+         |  ${graphId(graphNode)} [shape=plaintext label=${graphName(graphNode)}]
+         |}
+        """.stripMargin
```

Comment:
ToL

The last nine characters of this variable are `<newline><space><space><space><space><space><space><space><space>`.

This is a little bit neater. 

```scala
s"""|
    |subgraph $clusterName {
    |  style=filled;
    |  fillcolor=${graphFillColor(graphNode)};
    |  ${portNodes.mkString(sep="\n  ")}
    |  ${graphId(graphNode)} [shape=plaintext label=${graphName(graphNode)}]
    |}
    |""".stripMargin
```